python/ray/worker.py

```diff
@@ -423,11 +423,17 @@ def _try_store_and_register(self, object_id, value):
             # message.
             logger.info("The object with ID {} already exists "
                         "in the object store.".format(object_id))
-        except TypeError:
+        except TypeError as e:
             # TypeError can happen because one of the members of the object
             # may not be serializable for cloudpickle. So we need
             # these extra fallbacks here to start from the beginning.
             # Hopefully the object could have a `__reduce__` method.
+
+            # If the type error happens because a torch tensor was on the GPU
+            # and cannot be serialized due to that, we raise it to the user:
+            if "can't convert CUDA tensor to numpy" in e.args[0]:
```

Comment:
This seems like a strange place to handle it. Wouldn't it be better to just register a custom serializer for pytorch GPU tensors that raises an exception?